Rules/AvoidAssignmentToAutomaticVariable.cs

```diff
@@ -47,18 +47,21 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
             foreach (var assignmentStatementAst in assignmentStatementAsts)
             {
                 var variableExpressionAst = assignmentStatementAst.Find(testAst => testAst is VariableExpressionAst, searchNestedScriptBlocks: false) as VariableExpressionAst;
-                var variableName = variableExpressionAst.VariablePath.UserPath;
-                if (_readOnlyAutomaticVariables.Contains(variableName, StringComparer.OrdinalIgnoreCase))
+                if (variableExpressionAst != null)
```

Comment:
only this null check was added, all the code diff below is only the added indentation

Reply:
from a flow perspective, this could also be
```
if ( variableExpressionAst == null )
   continue
```
and the indenting is preserved :)